variables.tf

```
@@ -194,3 +194,33 @@ variable "dynamodb_table_name" {
   default     = null
   description = "Override the name of the DynamoDB table which defaults to using `module.dynamodb_table_label.id`"
 }
+
+variable "sse_algorithm" {
+  type        = string
+  default     = "AES256"
+  description = " The server-side encryption algorithm to use. Valid values are `AES256` and `aws:kms`"
+}
+
+variable "kms_key_description" {
+  type        = string
+  default     = null
+  description = "The description of the KMS key for the bucket encryption as viewed in AWS console."
+}
+
+variable "kms_key_enable_key_rotation" {
+  type        = bool
+  default     = null
+  description = "Specifies whether to enable key rotation."
+}
+
+variable "kms_key_deletion_window_in_days" {
+  type        = number
+  default     = null
+  description = "Number of days before AWS deletes the key."
+}
```

@vanastassiou You are correct that we want to convert this module to use the [s3-bucket module](https://github.com/cloudposse/terraform-aws-s3-bucket). As already mentioned, we place a high value on avoiding breaking changes. We also want to limit the scope of modules to make them easier to understand and maintain, which is why we do not include the ability to provision a KMS key in the s3-bucket module. Please use the [kms-key](https://github.com/cloudposse/terraform-aws-kms-key) module to create and manage the KMS key.

Please copy [the KMS key arn input](https://github.com/cloudposse/terraform-aws-s3-bucket/blob/caf2af9a03f947ae51348e675146f6475ca228a3/variables.tf#L68-L72) from the s3-bucket module:

```suggestion
variable "kms_master_key_arn" {
  type        = string
  default     = ""
  description = "The AWS KMS master key ARN used for the `SSE-KMS` encryption. This can only be used when you set the value of `sse_algorithm` as `aws:kms`. The default aws/s3 AWS KMS master key is used if this element is absent while the `sse_algorithm` is `aws:kms`"
}
```